addon/addon-test-support/@ember/test-helpers/-internal/build-registry.ts

```diff
@@ -72,46 +87,62 @@ const Owner = EmberObject.extend(RegistryProxyMixin, ContainerProxyMixin, {
  * @param {Object} resolver the resolver to use with the registry
  * @returns {Object} owner, container, registry
  */
-export default function (resolver: Resolver) {
-  let fallbackRegistry, registry, container;
-  let namespace = EmberObject.create({
-    // @ts-expect-error
-    Resolver: {
-      create() {
-        return resolver;
-      },
+export default function buildRegistry(resolver: Resolver) {
+  let namespace = new Application();
+  // @ts-ignore: this is actually the correcct type, but there was a typo in
```

Comment:
Similarly, many of these work correctly… on *some* versions of Ember. I used `ts-ignore` here and below to make it so that if there *is* an error, it is suppressed :sigh: but if not, it still has type safety.